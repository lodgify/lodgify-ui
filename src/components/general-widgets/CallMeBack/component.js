import React from 'react';
import PropTypes from 'prop-types';

import {
  CALL_ME_BACK,
  DATE,
  EMAIL,
  NAME,
  NOTES,
  PHONE,
  PROPERTY,
  SEND,
  TIME,
} from 'utils/default-strings';
import { Dropdown } from 'inputs/Dropdown';
import { Form } from 'collections/Form';
import { ICON_NAMES } from 'elements/Icon';
import { InputGroup } from 'collections/InputGroup';
import { PhoneInput } from 'inputs/PhoneInput';
import { SingleDatePicker } from 'inputs/SingleDatePicker';
import { TextArea } from 'inputs/TextArea';
import { TextInput } from 'inputs/TextInput';

/**
 * The standard widget for a user to request a call back.
 */
// eslint-disable-next-line jsdoc/require-jsdoc
export const Component = ({
  dateInputPlaceholder,
  emailInputLabel,
  errorMessage,
  headingText,
  nameInputLabel,
  notesInputLabel,
  onSubmit,
  phoneInputLabel,
  propertyInputLabel,
  propertyOptions,
  submitButtonText,
  successMessage,
  timeInputLabel,
  timeOptions,
  validation,
}) => (
  <Form
    autoComplete="off"
    errorMessage={errorMessage}
    headingText={headingText}
    onSubmit={onSubmit}
    submitButtonText={submitButtonText}
    successMessage={successMessage}
    validation={validation}
  >
    <InputGroup>
      <TextInput autoComplete="name" label={nameInputLabel} name="name" />
      <PhoneInput autoComplete="tel" label={phoneInputLabel} name="phone" />
    </InputGroup>
    <TextInput autoComplete="email" label={emailInputLabel} name="email" />
    <InputGroup>
      <SingleDatePicker name="date" placeholderText={dateInputPlaceholder} />
      <Dropdown
        icon={ICON_NAMES.CLOCK}
        label={timeInputLabel}
        name="time"
        options={timeOptions}
      />
    </InputGroup>
    <Dropdown
      label={propertyInputLabel}
      name="property"
      options={propertyOptions}
    />
    <TextArea label={notesInputLabel} name="notes" />
  </Form>
);

Component.displayName = 'CallMeBack';

Component.defaultProps = {
  dateInputPlaceholder: DATE,
  emailInputLabel: EMAIL,
  errorMessage: '',
  headingText: CALL_ME_BACK,
  nameInputLabel: NAME,
  notesInputLabel: NOTES,
  onSubmit: Function.prototype,
  phoneInputLabel: PHONE,
  propertyInputLabel: PROPERTY,
  submitButtonText: SEND,
  successMessage: '',
  timeInputLabel: TIME,
  validation: {},
};

Component.propTypes = {
  /** The placeholder for the date input. */
  dateInputPlaceholder: PropTypes.string,
  /** The label for the email input. */
  emailInputLabel: PropTypes.string,
  /** The message to display when the form has an error. */
  errorMessage: PropTypes.string,
  /** The text to display as a heading at the top of the widget. */
  headingText: PropTypes.string,
  /** The label for the name input. */
  nameInputLabel: PropTypes.string,
  /** The label for the notes input. */
  notesInputLabel: PropTypes.string,
  /** The function to call when the form is submitted
   *  @param {Object} values - The values of the inputs in the form.
   */
  onSubmit: PropTypes.func,
  /** The label for the phone input */
  phoneInputLabel: PropTypes.string,
  /** The label for the property input. */
  propertyInputLabel: PropTypes.string,
  /** The options which the user can select for the property field. */
  propertyOptions: PropTypes.arrayOf(
    PropTypes.shape({
      /** The visible text for the option. */
      text: PropTypes.string.isRequired,
      /** The underlying value for the option. */
      value: PropTypes.oneOfType([
        PropTypes.bool,
        PropTypes.number,
        PropTypes.string,
      ]),
    })
  ).isRequired,
  /** The form submit button text */
  submitButtonText: PropTypes.string,
  /** The message to display when the form is successful. */
  successMessage: PropTypes.string,
  /** The label for the time input. */
  timeInputLabel: PropTypes.string,
  /** The options which the user can select for the time field. */
  timeOptions: PropTypes.arrayOf(
    PropTypes.shape({
      /** The visible text for the option. */
      text: PropTypes.string.isRequired,
      /** The underlying value for the option. */
      value: PropTypes.oneOfType([
        PropTypes.bool,
        PropTypes.number,
        PropTypes.string,
      ]),
    })
  ).isRequired,
  /** Settings for validating inputs. Each value should match [the shape documented in `Form`](https://lodgify.github.io/lodgify-ui/#/Collections/Form) */
  validation: PropTypes.shape({
    date: PropTypes.object,
    email: PropTypes.object,
    name: PropTypes.object,
    notes: PropTypes.object,
    phone: PropTypes.object,
    property: PropTypes.object,
    time: PropTypes.object,
  }),
};
