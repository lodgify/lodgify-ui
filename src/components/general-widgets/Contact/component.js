import React from 'react';
import PropTypes from 'prop-types';

import { Form } from 'collections/Form';
import { InputGroup } from 'collections/InputGroup';
import { TextInput } from 'inputs/TextInput';
import { PhoneInput } from 'inputs/PhoneInput';
import { NumberInput } from 'inputs/NumberInput';
import { DateRangePicker } from 'inputs/DateRangePicker';
import { TextArea } from 'inputs/TextArea';
import { CaptchaInput } from 'inputs/CaptchaInput';
import { Dropdown } from 'inputs/Dropdown';
import {
  ARRIVAL,
  CONTACT,
  COMMENTS,
  DEPARTURE,
  EMAIL,
  GUESTS,
  NAME,
  PHONE,
  PROPERTY,
  ROOM,
  SECURITY_CODE,
  SEND,
} from 'utils/default-strings';

/**
 * The standard widget for a user contact an owner.
 */
// eslint-disable-next-line jsdoc/require-jsdoc
export const Component = ({
  arrivalDateInputLabel,
  captchaInputImage,
  captchaInputLabel,
  commentsInputLabel,
  dateRangePickerLocaleCode,
  departureDateInputLabel,
  emailInputLabel,
  errorMessage,
  guestsInputLabel,
  headingText,
  nameInputLabel,
  onChangeProperty,
  onSubmit,
  phoneInputLabel,
  propertyInputLabel,
  propertyOptions,
  roomInputLabel,
  roomOptions,
  submitButtonText,
  successMessage,
  validation,
}) => (
  <Form
    autoComplete="off"
    errorMessage={errorMessage}
    headingText={headingText}
    onSubmit={onSubmit}
    submitButtonText={submitButtonText}
    successMessage={successMessage}
    validation={validation}
  >
    <InputGroup>
      <TextInput autoComplete="name" label={nameInputLabel} name="name" />
      <PhoneInput autoComplete="tel" label={phoneInputLabel} name="phone" />
    </InputGroup>
    <TextInput autoComplete="email" label={emailInputLabel} name="email" />
    <InputGroup>
      <DateRangePicker
        endDatePlaceholderText={departureDateInputLabel}
        localeCode={dateRangePickerLocaleCode}
        name="dates"
        startDatePlaceholderText={arrivalDateInputLabel}
        width="eight"
      />
      <NumberInput
        label={guestsInputLabel}
        min={1}
        name="guests"
        width="four"
      />
    </InputGroup>
    <TextArea label={commentsInputLabel} name="comments" />
    {(roomOptions || propertyOptions) && (
      <InputGroup>
        {propertyOptions && (
          <Dropdown
            label={propertyInputLabel}
            name="property"
            onChange={onChangeProperty}
            options={propertyOptions}
          />
        )}
        {roomOptions && (
          <Dropdown label={roomInputLabel} name="room" options={roomOptions} />
        )}
      </InputGroup>
    )}
    <CaptchaInput
      image={captchaInputImage}
      label={captchaInputLabel}
      name="captcha"
    />
  </Form>
);

Component.displayName = 'Contact';

Component.defaultProps = {
  arrivalDateInputLabel: ARRIVAL,
  captchaInputLabel: SECURITY_CODE,
  commentsInputLabel: COMMENTS,
  dateRangePickerLocaleCode: undefined,
  departureDateInputLabel: DEPARTURE,
  emailInputLabel: EMAIL,
  errorMessage: '',
  guestsInputLabel: GUESTS,
  headingText: CONTACT,
  nameInputLabel: NAME,
  onChangeProperty: Function.prototype,
  onSubmit: Function.prototype,
  phoneInputLabel: PHONE,
  propertyInputLabel: PROPERTY,
  propertyOptions: null,
  roomInputLabel: ROOM,
  roomOptions: null,
  submitButtonText: SEND,
  successMessage: '',
  validation: {},
};

Component.propTypes = {
  /** The label for the arrival date input.*/
  arrivalDateInputLabel: PropTypes.string,
  /** The source url for the image to display. */
  captchaInputImage: PropTypes.string.isRequired,
  /** The label for the captcha input.*/
  captchaInputLabel: PropTypes.string,
  /** The label for the comments input.*/
  commentsInputLabel: PropTypes.string,
  /** The ISO 639-1 locale code which changes the format and language of days of the week and the months of the year in the date range picker. */
  dateRangePickerLocaleCode: PropTypes.string,
  /** The label for the departure date input.*/
  departureDateInputLabel: PropTypes.string,
  /** The label for the email input.*/
  emailInputLabel: PropTypes.string,
  /** The message to display when the form has an error. */
  errorMessage: PropTypes.string,
  /** The label for the guests input.*/
  guestsInputLabel: PropTypes.string,
  /** The text to display as a heading at the top of the widget. */
  headingText: PropTypes.string,
  /** The label for the name input.*/
  nameInputLabel: PropTypes.string,
  /** The function called when the property dropdown is changed.
   *  @param {String}        name - The name of the property dropdown field.
   *  @param {String|Number} value - The value of the property dropdown after the change
   */
  onChangeProperty: PropTypes.func,
  /** The function to call when the form is submitted
   *  @param {Object} values - The values of the inputs in the form.
   */
  onSubmit: PropTypes.func,
  /** The label for the phone input.*/
  phoneInputLabel: PropTypes.string,
  /** The label for the property input.*/
  propertyInputLabel: PropTypes.string,
  /** The options which the user can select for the property field. */
  propertyOptions: PropTypes.arrayOf(
    PropTypes.shape({
      /** The indent level of a property option. One of: 0, 1, 2, 3, 4, 5 */
      indent: PropTypes.oneOf([0, 1, 2, 3, 4, 5]),
      /** The visible text for the option. */
      text: PropTypes.string.isRequired,
      /** The underlying value for the option. */
      value: PropTypes.oneOfType([
        PropTypes.bool,
        PropTypes.number,
        PropTypes.string,
      ]),
    })
  ),
  /** The label for the room input.*/
  roomInputLabel: PropTypes.string,
  /** The options which the user can select for the room field. */
  roomOptions: PropTypes.arrayOf(
    PropTypes.shape({
      /** The visible text for the option. */
      text: PropTypes.string.isRequired,
      /** The underlying value for the option. */
      value: PropTypes.oneOfType([
        PropTypes.bool,
        PropTypes.number,
        PropTypes.string,
      ]),
    })
  ),
  /** The text to display on the submit button. */
  submitButtonText: PropTypes.string,
  /** The message to display when the form is successful. */
  successMessage: PropTypes.string,
  /** Settings for validating inputs. Each value should match [the shape documented in `Form`](https://lodgify.github.io/lodgify-ui/#/Collections/Form) */
  validation: PropTypes.shape({
    captcha: PropTypes.object,
    comments: PropTypes.object,
    dates: PropTypes.object,
    email: PropTypes.object,
    guests: PropTypes.object,
    name: PropTypes.object,
    phone: PropTypes.object,
    property: PropTypes.object,
    room: PropTypes.object,
  }),
};
