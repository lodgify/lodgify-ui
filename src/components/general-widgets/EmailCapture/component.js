import 'semantic-ui-styles/message.less';
import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { Message } from 'semantic-ui-react';

import { withResponsive } from 'utils/with-responsive';
import { Grid } from 'layout/Grid';
import { GridRow } from 'layout/GridRow';
import { GridColumn } from 'layout/GridColumn';
import { Heading } from 'typography/Heading';
import { ShowOn } from 'layout/ShowOn';
import { TextInput } from 'inputs/TextInput';
import { Button } from 'elements/Button';
import { Divider } from 'elements/Divider';
import { HorizontalGutters } from 'layout/HorizontalGutters';
import {
  ACCEPT_PRIVACY_POLICY,
  SUBSCRIBE,
  SUBSCRIBE_TO_OUR_NEWSLETTER,
  YOUR_EMAIL,
} from 'utils/default-strings';

import { getPrivacyCheckboxMarkup } from './utils/getPrivacyCheckboxMarkup';

/**
 * An email capture displays a form inviting a user to submit an email address.
 */
// eslint-disable-next-line jsdoc/require-jsdoc
const Component = ({
  buttonText,
  emailInputError,
  emailInputLabel,
  emailInputValue,
  errorMessage,
  headingText,
  isPrivacyConsentInputChecked,
  isPrivacyConsentRequired,
  isUserOnMobile,
  onChangeEmailInput,
  onClickButton,
  onClickPrivacyConsentInput,
  privacyConsentInputError,
  privacyConsentLabelLinkText,
  privacyConsentLabelLinkUrl,
  privacyConsentLabelText,
  successMessage,
}) => (
  <HorizontalGutters>
    <Divider />
    <Grid>
      <GridRow verticalAlign="middle">
        {!errorMessage ? (
          <Fragment>
            <GridColumn computer={5} mobile={12} tablet={12} textAlign="center">
              <Heading hasMargin={false} size="small">
                {headingText}
              </Heading>
              <ShowOn
                mobile
                parent={Divider}
                parentProps={{ size: 'small' }}
                tablet
              />
            </GridColumn>
            <GridColumn computer={4} mobile={12} tablet={7}>
              <TextInput
                error={emailInputError}
                isFluid
                label={emailInputLabel}
                onChange={onChangeEmailInput}
                value={emailInputValue}
              />
              <ShowOn mobile parent={Divider} parentProps={{ size: 'small' }} />
            </GridColumn>
            <GridColumn computer={3} mobile={12} tablet={5}>
              {isUserOnMobile && isPrivacyConsentRequired && (
                <Fragment>
                  {getPrivacyCheckboxMarkup(
                    privacyConsentInputError,
                    isPrivacyConsentInputChecked,
                    privacyConsentLabelText,
                    privacyConsentLabelLinkUrl,
                    privacyConsentLabelLinkText,
                    onClickPrivacyConsentInput
                  )}
                  <Divider />
                </Fragment>
              )}
              {!successMessage ? (
                <Button isFluid isRounded onClick={onClickButton}>
                  {buttonText}
                </Button>
              ) : (
                <Message content={successMessage} success />
              )}
            </GridColumn>
          </Fragment>
        ) : (
          <GridColumn width={12}>
            <Message content={errorMessage} error />
          </GridColumn>
        )}
      </GridRow>
      {!errorMessage && !isUserOnMobile && isPrivacyConsentRequired && (
        <GridRow>
          <GridColumn computer={5} />
          <GridColumn computer={4} mobile={12} tablet={12}>
            {getPrivacyCheckboxMarkup(
              privacyConsentInputError,
              isPrivacyConsentInputChecked,
              privacyConsentLabelText,
              privacyConsentLabelLinkUrl,
              privacyConsentLabelLinkText,
              onClickPrivacyConsentInput
            )}
          </GridColumn>
        </GridRow>
      )}
    </Grid>
    <Divider />
  </HorizontalGutters>
);

export const ComponentWithResponsive = withResponsive(Component);

Component.displayName = 'EmailCapture';

Component.defaultProps = {
  buttonText: SUBSCRIBE,
  emailInputError: false,
  emailInputLabel: YOUR_EMAIL,
  emailInputValue: undefined,
  errorMessage: null,
  headingText: SUBSCRIBE_TO_OUR_NEWSLETTER,
  isPrivacyConsentInputChecked: false,
  isPrivacyConsentRequired: false,
  onChangeEmailInput: Function.prototype,
  onClickPrivacyConsentInput: Function.prototype,
  onClickButton: Function.prototype,
  privacyConsentInputError: undefined,
  privacyConsentLabelLinkText: undefined,
  privacyConsentLabelLinkUrl: undefined,
  privacyConsentLabelText: ACCEPT_PRIVACY_POLICY,
  successMessage: null,
};

Component.propTypes = {
  /** The text to display on the button. */
  buttonText: PropTypes.string,
  /** An error message to display in place of the component. */
  emailInputError: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
  /** The visible label for the text input. */
  emailInputLabel: PropTypes.string,
  /** The value of the text input. */
  emailInputValue: PropTypes.string,
  /** Is the text input in an error state. */
  errorMessage: PropTypes.string,
  /** The text to display as a heading. */
  headingText: PropTypes.string,
  /** The value of the privacy consent checkbox. */
  isPrivacyConsentInputChecked: PropTypes.bool,
  /** Displays a privacy consent checkbox. */
  isPrivacyConsentRequired: PropTypes.bool,
  /**
   * Is the user on a mobile device.
   * Provided by `withResponsive` so ignored in the styleguide.
   * @ignore
   */
  isUserOnMobile: PropTypes.bool.isRequired,
  /** The function called when the text input is changed.
   *  @param {String} name
   *  @param {String} value
   */
  onChangeEmailInput: PropTypes.func,
  /** The function called when the button is clicked.
   *  @param {String} name
   *  @param {String} value
   */
  onClickButton: PropTypes.func,
  /** The function called when the privacy consent checkbox is clicked.
   *  @param {Object} event
   */
  onClickPrivacyConsentInput: PropTypes.func,
  /** The text to display in the privacy consent error message. */
  privacyConsentInputError: PropTypes.string,
  /** The text to display as the link next to the privacy consent checkbox. */
  privacyConsentLabelLinkText: PropTypes.string,
  /** The location of the link next to the privacy consent checkbox. */
  privacyConsentLabelLinkUrl: PropTypes.string,
  /** The text to display next to the privacy consent checkbox. */
  privacyConsentLabelText: PropTypes.node,
  /** A success message to display in place of the button. */
  successMessage: PropTypes.string,
};
