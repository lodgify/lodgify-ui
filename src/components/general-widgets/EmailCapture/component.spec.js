import React from 'react';
import { mount } from 'enzyme';
import { expectComponentToHaveDisplayName } from '@lodgify/enzyme-jest-expect-helpers';

import { Component as EmailCapture } from './component';

const getEmailCapture = props => mount(<EmailCapture {...props} />);

describe('<EmailCapture />', () => {
  it('should render the right structure', () => {
    const wrapper = getEmailCapture();

    expect(wrapper).toMatchSnapshot();
  });

  describe('if `props.errorMessage` is passed', () => {
    it('should render the right structure', () => {
      const wrapper = getEmailCapture({ errorMessage: 'Nurrrrrrr' });

      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('if `props.successMessage` is passed', () => {
    it('should render the right structure', () => {
      const wrapper = getEmailCapture({ successMessage: 'Yurrrrrrr' });

      expect(wrapper).toMatchSnapshot();
    });
  });

  it('should have `displayName` EmailCapture', () => {
    expectComponentToHaveDisplayName(EmailCapture, 'EmailCapture');
  });
});
