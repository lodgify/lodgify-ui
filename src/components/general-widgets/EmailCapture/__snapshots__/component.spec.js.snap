// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmailCapture /> if \`props.errorMessage\` is passed should render the right structure 1`] = `
<EmailCapture
  buttonText="Subscribe"
  errorMessage="Nurrrrrrr"
  headingText="Subscribe to our newsletter"
  inputError={false}
  inputLabel="Your email"
  onChangeInput={[Function]}
  onClickButton={[Function]}
  successMessage={null}
>
  <HorizontalGutters>
    <Container>
      <div
        className="ui container"
      >
        <Divider
          className=""
          hasLine={false}
          size="medium"
        >
          <Divider
            className=""
            hidden={true}
            section={false}
          >
            <div
              className="ui hidden divider"
            />
          </Divider>
        </Divider>
        <Grid
          areColumnsCentered={false}
          isStackable={false}
        >
          <Grid
            centered={false}
            stackable={false}
          >
            <div
              className="ui grid"
            >
              <GridRow
                horizontalAlignContent="left"
                verticalAlign="middle"
              >
                <GridRow
                  textAlign="left"
                  verticalAlign="middle"
                >
                  <div
                    className="left aligned middle aligned row"
                  >
                    <GridColumn
                      verticalAlignContent="top"
                      width={12}
                    >
                      <GridColumn
                        verticalAlign="top"
                        width={12}
                      >
                        <div
                          className="top aligned twelve wide column"
                        >
                          <Message
                            content="Nurrrrrrr"
                            error={true}
                          >
                            <div
                              className="ui error message"
                            >
                              <MessageContent>
                                <div
                                  className="content"
                                >
                                  <p
                                    key="Nurrrrrrr"
                                  >
                                    Nurrrrrrr
                                  </p>
                                </div>
                              </MessageContent>
                            </div>
                          </Message>
                        </div>
                      </GridColumn>
                    </GridColumn>
                  </div>
                </GridRow>
              </GridRow>
            </div>
          </Grid>
        </Grid>
        <Divider
          className=""
          hasLine={false}
          size="medium"
        >
          <Divider
            className=""
            hidden={true}
            section={false}
          >
            <div
              className="ui hidden divider"
            />
          </Divider>
        </Divider>
      </div>
    </Container>
  </HorizontalGutters>
</EmailCapture>
`;

exports[`<EmailCapture /> if \`props.successMessage\` is passed should render the right structure 1`] = `
<EmailCapture
  buttonText="Subscribe"
  errorMessage={null}
  headingText="Subscribe to our newsletter"
  inputError={false}
  inputLabel="Your email"
  onChangeInput={[Function]}
  onClickButton={[Function]}
  successMessage="Yurrrrrrr"
>
  <HorizontalGutters>
    <Container>
      <div
        className="ui container"
      >
        <Divider
          className=""
          hasLine={false}
          size="medium"
        >
          <Divider
            className=""
            hidden={true}
            section={false}
          >
            <div
              className="ui hidden divider"
            />
          </Divider>
        </Divider>
        <Grid
          areColumnsCentered={false}
          isStackable={false}
        >
          <Grid
            centered={false}
            stackable={false}
          >
            <div
              className="ui grid"
            >
              <GridRow
                horizontalAlignContent="left"
                verticalAlign="middle"
              >
                <GridRow
                  textAlign="left"
                  verticalAlign="middle"
                >
                  <div
                    className="left aligned middle aligned row"
                  >
                    <GridColumn
                      computer={5}
                      mobile={12}
                      tablet={12}
                      textAlign="center"
                      verticalAlignContent="top"
                      width={null}
                    >
                      <GridColumn
                        computer={5}
                        mobile={12}
                        tablet={12}
                        textAlign="center"
                        verticalAlign="top"
                        width={null}
                      >
                        <div
                          className="center aligned top aligned five wide computer twelve wide mobile twelve wide tablet column"
                        >
                          <Heading
                            className={null}
                            hasMargin={false}
                            isColorInverted={false}
                            size="small"
                            textAlign={null}
                          >
                            <Header
                              as="h4"
                              className="has-no-margin"
                              inverted={false}
                              textAlign={null}
                            >
                              <h4
                                className="ui header has-no-margin"
                              >
                                Subscribe to our newsletter
                              </h4>
                            </Header>
                          </Heading>
                          <ShowOn
                            computer={false}
                            mobile={true}
                            parent={[Function]}
                            parentProps={
                              Object {
                                "size": "small",
                              }
                            }
                            tablet={true}
                            widescreen={false}
                          >
                            <Divider
                              className="show-on-mobile show-on-tablet"
                              hasLine={false}
                              size="small"
                            >
                              <Divider
                                className="show-on-mobile show-on-tablet is-size-small"
                                hidden={true}
                                section={false}
                              >
                                <div
                                  className="ui hidden divider show-on-mobile show-on-tablet is-size-small"
                                />
                              </Divider>
                            </Divider>
                          </ShowOn>
                        </div>
                      </GridColumn>
                    </GridColumn>
                    <GridColumn
                      computer={4}
                      mobile={12}
                      tablet={7}
                      verticalAlignContent="top"
                      width={null}
                    >
                      <GridColumn
                        computer={4}
                        mobile={12}
                        tablet={7}
                        verticalAlign="top"
                        width={null}
                      >
                        <div
                          className="top aligned four wide computer twelve wide mobile seven wide tablet column"
                        >
                          <TextInput
                            autoComplete={null}
                            error={false}
                            isFluid={true}
                            isValid={false}
                            label="Your email"
                            maxCharacters={null}
                            name=""
                            onBlur={[Function]}
                            onChange={[Function]}
                            type="text"
                          >
                            <InputController
                              adaptOnChangeEvent={[Function]}
                              error={false}
                              icon={null}
                              initialValue=""
                              inputOnChangeFunctionName="onChange"
                              isFluid={true}
                              isFocused={false}
                              isValid={false}
                              label="Your email"
                              mapValueToProps={[Function]}
                              name=""
                              onChange={[Function]}
                            >
                              <Input
                                className=""
                                fluid={true}
                                iconPosition={null}
                                type="text"
                              >
                                <div
                                  className="ui fluid input"
                                >
                                  <input
                                    autoComplete={null}
                                    key=".0"
                                    maxLength={null}
                                    name=""
                                    onBlur={[Function]}
                                    onChange={[Function]}
                                    type="text"
                                    value=""
                                  />
                                  <label
                                    key=".3"
                                    onClick={[Function]}
                                  >
                                    Your email
                                  </label>
                                </div>
                              </Input>
                            </InputController>
                          </TextInput>
                          <ShowOn
                            computer={false}
                            mobile={true}
                            parent={[Function]}
                            parentProps={
                              Object {
                                "size": "small",
                              }
                            }
                            tablet={false}
                            widescreen={false}
                          >
                            <Divider
                              className="show-on-mobile"
                              hasLine={false}
                              size="small"
                            >
                              <Divider
                                className="show-on-mobile is-size-small"
                                hidden={true}
                                section={false}
                              >
                                <div
                                  className="ui hidden divider show-on-mobile is-size-small"
                                />
                              </Divider>
                            </Divider>
                          </ShowOn>
                        </div>
                      </GridColumn>
                    </GridColumn>
                    <GridColumn
                      computer={3}
                      mobile={12}
                      tablet={5}
                      verticalAlignContent="top"
                      width={null}
                    >
                      <GridColumn
                        computer={3}
                        mobile={12}
                        tablet={5}
                        verticalAlign="top"
                        width={null}
                      >
                        <div
                          className="top aligned three wide computer twelve wide mobile five wide tablet column"
                        >
                          <Message
                            content="Yurrrrrrr"
                            success={true}
                          >
                            <div
                              className="ui success message"
                            >
                              <MessageContent>
                                <div
                                  className="content"
                                >
                                  <p
                                    key="Yurrrrrrr"
                                  >
                                    Yurrrrrrr
                                  </p>
                                </div>
                              </MessageContent>
                            </div>
                          </Message>
                        </div>
                      </GridColumn>
                    </GridColumn>
                  </div>
                </GridRow>
              </GridRow>
            </div>
          </Grid>
        </Grid>
        <Divider
          className=""
          hasLine={false}
          size="medium"
        >
          <Divider
            className=""
            hidden={true}
            section={false}
          >
            <div
              className="ui hidden divider"
            />
          </Divider>
        </Divider>
      </div>
    </Container>
  </HorizontalGutters>
</EmailCapture>
`;

exports[`<EmailCapture /> should render the right structure 1`] = `
<EmailCapture
  buttonText="Subscribe"
  errorMessage={null}
  headingText="Subscribe to our newsletter"
  inputError={false}
  inputLabel="Your email"
  onChangeInput={[Function]}
  onClickButton={[Function]}
  successMessage={null}
>
  <HorizontalGutters>
    <Container>
      <div
        className="ui container"
      >
        <Divider
          className=""
          hasLine={false}
          size="medium"
        >
          <Divider
            className=""
            hidden={true}
            section={false}
          >
            <div
              className="ui hidden divider"
            />
          </Divider>
        </Divider>
        <Grid
          areColumnsCentered={false}
          isStackable={false}
        >
          <Grid
            centered={false}
            stackable={false}
          >
            <div
              className="ui grid"
            >
              <GridRow
                horizontalAlignContent="left"
                verticalAlign="middle"
              >
                <GridRow
                  textAlign="left"
                  verticalAlign="middle"
                >
                  <div
                    className="left aligned middle aligned row"
                  >
                    <GridColumn
                      computer={5}
                      mobile={12}
                      tablet={12}
                      textAlign="center"
                      verticalAlignContent="top"
                      width={null}
                    >
                      <GridColumn
                        computer={5}
                        mobile={12}
                        tablet={12}
                        textAlign="center"
                        verticalAlign="top"
                        width={null}
                      >
                        <div
                          className="center aligned top aligned five wide computer twelve wide mobile twelve wide tablet column"
                        >
                          <Heading
                            className={null}
                            hasMargin={false}
                            isColorInverted={false}
                            size="small"
                            textAlign={null}
                          >
                            <Header
                              as="h4"
                              className="has-no-margin"
                              inverted={false}
                              textAlign={null}
                            >
                              <h4
                                className="ui header has-no-margin"
                              >
                                Subscribe to our newsletter
                              </h4>
                            </Header>
                          </Heading>
                          <ShowOn
                            computer={false}
                            mobile={true}
                            parent={[Function]}
                            parentProps={
                              Object {
                                "size": "small",
                              }
                            }
                            tablet={true}
                            widescreen={false}
                          >
                            <Divider
                              className="show-on-mobile show-on-tablet"
                              hasLine={false}
                              size="small"
                            >
                              <Divider
                                className="show-on-mobile show-on-tablet is-size-small"
                                hidden={true}
                                section={false}
                              >
                                <div
                                  className="ui hidden divider show-on-mobile show-on-tablet is-size-small"
                                />
                              </Divider>
                            </Divider>
                          </ShowOn>
                        </div>
                      </GridColumn>
                    </GridColumn>
                    <GridColumn
                      computer={4}
                      mobile={12}
                      tablet={7}
                      verticalAlignContent="top"
                      width={null}
                    >
                      <GridColumn
                        computer={4}
                        mobile={12}
                        tablet={7}
                        verticalAlign="top"
                        width={null}
                      >
                        <div
                          className="top aligned four wide computer twelve wide mobile seven wide tablet column"
                        >
                          <TextInput
                            autoComplete={null}
                            error={false}
                            isFluid={true}
                            isValid={false}
                            label="Your email"
                            maxCharacters={null}
                            name=""
                            onBlur={[Function]}
                            onChange={[Function]}
                            type="text"
                          >
                            <InputController
                              adaptOnChangeEvent={[Function]}
                              error={false}
                              icon={null}
                              initialValue=""
                              inputOnChangeFunctionName="onChange"
                              isFluid={true}
                              isFocused={false}
                              isValid={false}
                              label="Your email"
                              mapValueToProps={[Function]}
                              name=""
                              onChange={[Function]}
                            >
                              <Input
                                className=""
                                fluid={true}
                                iconPosition={null}
                                type="text"
                              >
                                <div
                                  className="ui fluid input"
                                >
                                  <input
                                    autoComplete={null}
                                    key=".0"
                                    maxLength={null}
                                    name=""
                                    onBlur={[Function]}
                                    onChange={[Function]}
                                    type="text"
                                    value=""
                                  />
                                  <label
                                    key=".3"
                                    onClick={[Function]}
                                  >
                                    Your email
                                  </label>
                                </div>
                              </Input>
                            </InputController>
                          </TextInput>
                          <ShowOn
                            computer={false}
                            mobile={true}
                            parent={[Function]}
                            parentProps={
                              Object {
                                "size": "small",
                              }
                            }
                            tablet={false}
                            widescreen={false}
                          >
                            <Divider
                              className="show-on-mobile"
                              hasLine={false}
                              size="small"
                            >
                              <Divider
                                className="show-on-mobile is-size-small"
                                hidden={true}
                                section={false}
                              >
                                <div
                                  className="ui hidden divider show-on-mobile is-size-small"
                                />
                              </Divider>
                            </Divider>
                          </ShowOn>
                        </div>
                      </GridColumn>
                    </GridColumn>
                    <GridColumn
                      computer={3}
                      mobile={12}
                      tablet={5}
                      verticalAlignContent="top"
                      width={null}
                    >
                      <GridColumn
                        computer={3}
                        mobile={12}
                        tablet={5}
                        verticalAlign="top"
                        width={null}
                      >
                        <div
                          className="top aligned three wide computer twelve wide mobile five wide tablet column"
                        >
                          <Button
                            hasShadow={false}
                            icon={null}
                            isCompact={false}
                            isDisabled={false}
                            isFluid={true}
                            isFormSubmit={false}
                            isLoading={false}
                            isPositionedRight={false}
                            isRounded={true}
                            isSecondary={false}
                            onClick={[Function]}
                            size={null}
                          >
                            <Button
                              as="button"
                              circular={true}
                              className=""
                              compact={false}
                              disabled={false}
                              floated="left"
                              fluid={true}
                              loading={false}
                              onClick={[Function]}
                              secondary={false}
                              size={null}
                              type="button"
                            >
                              <button
                                className="ui circular fluid left floated button"
                                onClick={[Function]}
                                role="button"
                                type="button"
                              >
                                Subscribe
                              </button>
                            </Button>
                          </Button>
                        </div>
                      </GridColumn>
                    </GridColumn>
                  </div>
                </GridRow>
              </GridRow>
            </div>
          </Grid>
        </Grid>
        <Divider
          className=""
          hasLine={false}
          size="medium"
        >
          <Divider
            className=""
            hidden={true}
            section={false}
          >
            <div
              className="ui hidden divider"
            />
          </Divider>
        </Divider>
      </div>
    </Container>
  </HorizontalGutters>
</EmailCapture>
`;
